src/agoradatatools/process.py

```diff
@@ -16,59 +17,63 @@
-        df = datasets["overall_scores"]
-        return transform.transform_overall_scores(df=df)
-    if dataset_name == "distribution_data":
-        return transform.transform_distribution_data(
-            datasets=datasets,
-            overall_max_score=dataset_obj["custom_transformations"][
-                "overall_max_score"
-            ],
-            genetics_max_score=dataset_obj["custom_transformations"][
-                "genetics_max_score"
-            ],
-            omics_max_score=dataset_obj["custom_transformations"]["omics_max_score"],
-        )
-    if dataset_name == "team_info":
-        return transform.transform_team_info(datasets=datasets)
-    if dataset_name == "rnaseq_differential_expression":
-        return transform.transform_rnaseq_differential_expression(datasets=datasets)
-    if dataset_name == "gene_info":
-        return transform.transform_gene_info(
-            datasets=datasets,
-            adjusted_p_value_threshold=dataset_obj["custom_transformations"][
-                "adjusted_p_value_threshold"
-            ],
-            protein_level_threshold=dataset_obj["custom_transformations"][
-                "protein_level_threshold"
-            ],
-        )
-    if dataset_name == "rna_distribution_data":
-        return transform.transform_rna_distribution_data(datasets=datasets)
-    if dataset_name == "proteomics_distribution_data":
-        return transform.transform_proteomics_distribution_data(datasets=datasets)
-    if dataset_name in ["proteomics", "proteomics_tmt", "proteomics_srm"]:
-        df = datasets[dataset_name]
-        return transform.transform_proteomics(df=df)
-    if dataset_name in ["biomarkers", "pathology"]:
-        return transform.immunohisto_transform(
-            datasets=datasets, dataset_name=dataset_name
-        )
-    if dataset_name == "model_details":
-        return transform.transform_model_details(datasets=datasets)
+    if isinstance(function_info, str):
+        function_name = function_info
+        config_defined_params = {}
     else:
-        return None
+        # Retrieve the function name and its parameters
+        # Assumes a single function in the dict
+        if len(function_info.items()) != 1:
+            warnings.warn(
+                "Please provide a single custom transformation function in the configuration file. Only the first function will be used if multiple are provided."
+            )
+        function_name, config_defined_params = next(iter(function_info.items()))
+    if not hasattr(transform, function_name):
+        raise AttributeError(
+            f"Function {function_name} not found in the transform module. Please provide the correct function name."
+        )
+    
+    retrieved_function = getattr(transform, function_name)
+    sig = inspect.signature(retrieved_function)
+    parameters = {}
+    # Map known inputs
+    for name, _ in sig.parameters.items():
+        if name == "df":
```

Comment:
Comment for future: I wonder if we should enforce some standardization on our transform functions so we don't end up adding more to this if-statement. i.e. the transform always needs to take either a Data.Frame, a list of Data.Frames, and/or a string that is named one of these things?

Reply:
see my [suggestion](https://github.com/Sage-Bionetworks/agora-data-tools/pull/176#discussion_r2073592913) which could help out if more data types are needed. We could also enforce that only those we support are allowed though.

Reply:
I just saw your suggestion, yeah that makes it a lot cleaner.